include/deal.II/lac/sparse_matrix_ez.h

```diff
@@ -683,6 +683,17 @@ class SparseMatrixEZ : public Subscriptor
   const_iterator begin (const size_type r) const;
 
   /**
+   * Iterator on diagonal entry of row @p r.
+   *
+   * @warning If the current row has no diagonal element, then the returned
+   * iterator will be same as end(r), i.e., an iterator points to first element
+   * of next row or same place behind end of the entire matrix.
+   * So, do not dereference or query information to the returned iterator blindly
+   * unless you can make sure your matrix do have a diagonal element in row @p r.
```

Comment:
This  return value is necessary for loop like
```
for( it=diagonal(r); it != end(r);++it)
```
or
```
for( it=begin(r); it != diagonal(r);++it)
```
to perform as expected when there is no diagonal element exists. If ```begin(r)``` returns iterator on the end of entire matrix, these loops will go too far.

And, when ```r``` is the last row, ```end(r)``` and ```end()``` returns the same thing -- a pointer immediately behind last valid element of the entire matrix -- which is ready for access violation.

So, generally, the Accessor and Iterator class is not safe. I just not make situation even worse, I think.